api/nodes/views.py

```diff
@@ -1745,3 +1748,115 @@ def perform_create(self, serializer):
         serializer.validated_data['target'] = node
         serializer.validated_data['node'] = node
         serializer.save()
+
+class NodeInstitutionDetail(JSONAPIBaseView, generics.RetrieveAPIView, NodeMixin):
+    """ Detail of the one primary_institution a node has, if any. Returns NotFound
+    if the node does not have a primary_instituion.
+
+    ##Attributes
+
+    OSF Institutions have the "institutions" `type`.
+
+        name           type               description
+        -------------------------------------------------------------------------
+        name           string             title of the institution
+        id             string             unique identifier in the OSF
+        logo_path      string             a path to the institution's static logo
+
+
+
+    #This Request/Response
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        base_permissions.TokenHasScope,
+        AdminOrPublic
+    )
+
+    required_read_scopes = [CoreScopes.NODE_BASE_READ, CoreScopes.INSTITUTION_READ]
+    required_write_scopes = [CoreScopes.NULL]
+    serializer_class = InstitutionSerializer
+
+    model = Institution
+    view_category = 'nodes'
+    view_name = 'node-institution-detail'
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        node = self.get_node()
+        if not node.primary_institution:
+            raise NotFound
+        return node.primary_institution
+
+
+class NodeInstitutionRelationship(JSONAPIBaseView, generics.RetrieveUpdateAPIView, NodeMixin):
+    """ Relationship Endpoint for Node -> Instituion Relationship
+
+    Used to set the primary_institution of a node to an institution
```

Comment:
This is a bit confusing because in InstitutionsList for example, type is "institutions" and for this endpoint, type is "institution".  This either needs to clearly be spelled out or we should just pick one. My feelings are that we shouldn't be worrying about single versus plural in the "type", this is really to differentiate is this a "users" type, or a "nodes" type, for example.   I feel the institutions should have "institutions" type across the board, plural.